main.go

```diff
@@ -307,6 +313,54 @@ func main() {
 		}
 	}()
 
+	Ip64PacketCount := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ip64_packets_count",
+			Help: "Packet count for each reason and protocol",
+		},
+		[]string{"status", "protocol"},
+	)
+	Ip46PacketCount := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ip46_packets_count",
+			Help: "Packet count for each reason and protocol",
+		},
+		[]string{"status", "protocol"},
+	)
+
+	prometheus.MustRegister(Ip64PacketCount)
+	prometheus.MustRegister(Ip46PacketCount)
+
+	Ip64map := metrics.LoadMap("ip64_metrics")
+	Ip46map := metrics.LoadMap("ip46_metrics")
+
+	defer func() {
+		err = Ip64map.Close()
+		if err != nil {
+			klog.Infof("error closing map object")
+		}
+	}()
+	defer func() {
+		err = Ip46map.Close()
+		if err != nil {
+			klog.Infof("error closing map object")
+		}
+	}()
+
+	metricsTicker := time.NewTicker(metricInterval)
+	defer metricsTicker.Stop()
+
+	klog.Infof("Start auto sync metrics")
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for range metricsTicker.C {
+			metrics.ReadAndUpdatePacketCount(Ip64map, Ip64PacketCount)
+			metrics.ReadAndUpdatePacketCount(Ip46map, Ip46PacketCount)
+		}
+	}()
```

Comment:
you are creating a WaitGroup here but you are not waiting for it anywhere, you can just return when the context is done and update the metrics every tick


```suggestion
	klog.Infof("Start auto sync metrics")
        go func() {
            for {
                select {
                case <-metricsTicker.C:
                    metrics.ReadAndUpdatePacketCount(Ip64map, Ip64PacketCount)
                    metrics.ReadAndUpdatePacketCount(Ip46map, Ip46PacketCount)
                case <-ctx.Done(): 
                    return
                }
            }
        }()
```